src/plugins/data/public/index_patterns/lib/get_from_saved_object.ts

```diff
@@ -18,15 +18,17 @@
  */
 
 import { get } from 'lodash';
+import { IIndexPattern } from '../..';
 
-export function getFromSavedObject(savedObject: any) {
+export function getFromSavedObject(savedObject: any): IIndexPattern | undefined {
```

Comment:
Why did you have to explicitly define the return type here?
Could have cast the returned result at the caller?

Reply:
Because the object no longer contained all the properties on the interface. I was just confused as to when I should use `IIndexPattern` over `PublicMethodsOf<IndexPatterns>`.

I'll but back the `IIndexPattern` interface and remove these changes.

Reply:
d143c22